asv_bench/benchmarks/utils.py

```diff
@@ -447,3 +487,30 @@ def get_shape_id(shape: tuple) -> str:
     str
     """
     return "_".join([str(element) for element in shape])
+
+
+def prepare_io_data(test_filename: str, data_type: str):
+    """
+    Prepare data for IO tests with caching.
+
+    Parameters
+    ----------
+    test_filename : str
+        Unique file identifier that is used to distinguish data
+        for different tests.
+    data_type : {"int", "str_int", "true_false_int"}
+        Type of data generation.
+
+    Returns
+    -------
+    test_filenames : dict
+        Dcitionary that maps dataset shape to the file on disk.
+    """
+    test_filenames = {}
+    for shape in UNARY_OP_DATA_SIZE[ASV_DATASET_SIZE]:
```

Comment:
It is dangerous enough to use a closure in a situation where the variable being closed is defined in several places.

```suggestion
def prepare_io_data(test_filename: str, data_type: str, shapes: list):
    """
    Prepare data for IO tests with caching.

    Parameters
    ----------
    test_filename : str
        Unique file identifier that is used to distinguish data
        for different tests.
    data_type : {"int", "str_int", "true_false_int"}
        Type of data generation.

    Returns
    -------
    test_filenames : dict
        Dcitionary that maps dataset shape to the file on disk.
    """
    test_filenames = {}
    for shape in shapes:
```